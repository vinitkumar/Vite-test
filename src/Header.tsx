
function Header() {
    return (
        <>
          <h1>Hello, I am a fish Header</h1>
        </>
    );
}

export default Header;
