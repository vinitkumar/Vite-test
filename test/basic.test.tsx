import React from "react";
import renderer from "react-test-renderer";
import Header from "../src/Header";



function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}


test('Check the App components', () => {
    const AppComponent = renderer.create(
        <Header/>
    );

    let tree = toJson(AppComponent);
    console.log(tree);
    expect(tree).toMatchSnapshot();
});


