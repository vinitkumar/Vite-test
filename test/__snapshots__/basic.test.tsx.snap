// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Check the App components 1`] = `
<h1>
  Hello, I am a fish Header
</h1>
`;
